### DISCUSSION TITLE
Porting a new PDK

### DISCUSSION THREAD

**Discussion by jarvis-lorry** [siliconcompiler/siliconcompiler#1056] [Help (Q&A)] [answered]
Is it possible to port a new PDK into SiliconCompiler, a PDK not included in the list of supported open PDKs?

**Comment by nmoroze** [comment, marked as answer]
Yes, this should be possible! You can refer to this page in the [docs](https://docs.siliconcompiler.com/en/latest/user_guide/pdks.html) and use the open PDKs as a model. Feel free to let us know if anything is unclear or if you get stuck on anything in particular!

**Reply by adshay17** [reply]
The link to docs throws an error, are you referring to https://docs.siliconcompiler.com/en/latest/development_guide/pdks.html 
???

**Reply by gadfort** [reply]
@adshay17 yes, that is the correct page

**Reply by adshay17** [reply]
I'm working through connecting a separate PDK that is not included in the SiliconCompiler targets into a flow. I worked through using the [setup instructions](https://docs.siliconcompiler.com/en/stable/development_guide/pdks.html) provided by the siliconcompiler documentation and used the [lamdapdk example scripts](https://github.com/siliconcompiler/lambdapdk) for open source PDK's as reference to connect my own. I created pdk and stdlib objects and attempted to call them via chip.use(setup_pdk()) and chip.use(setup_stdlib()). For the third component I tried to use the [synflow siliconcompiler pre built flow ](https://docs.siliconcompiler.com/en/stable/reference_manual/predef_modules/flows.html#flows-synflow-ref)to run a simple flow with my connected pdk and library. I am getting an error (| ERROR    | job0 | - | - | synflow flowgraph contains errors and cannot be run.) and am having trouble identifying where to go to debug this. I am unsure if I am configuring things wrong all together or if there's something else going on. I have access to all of my pdks data and library files, opensource tools, and siliconcompiler packages.

**Reply by gadfort** [reply]
@adshay17 somewhere in the log it should specify anything that was identified as an issue. (I'm assuming that you are attempting to use yosys as the synthesis tool). It would be hard to debug without access to the logs or the code. If the issue persists, please open an issue and provide what you can.
